packages/pyright-internal/src/analyzer/tracePrinter.ts

```diff
@@ -22,36 +22,33 @@ export type PrintableType = ParseNode | Declaration | Symbol | Type | undefined;
 
 export interface TracePrinter {
     print(o: PrintableType): string;
-    printFileOrModuleName(filePathOrModule: string | AbsoluteModuleDescriptor): string;
+    printFileOrModuleName(fileUriOrModule: Uri | AbsoluteModuleDescriptor): string;
 }
 
-export function createTracePrinter(roots: string[]): TracePrinter {
+export function createTracePrinter(roots: Uri[]): TracePrinter {
     function wrap(value: string | undefined, ch = "'") {
         return value ? `${ch}${value}${ch}` : '';
     }
 
     // Sort roots in desc order so that we compare longer path first
     // when getting relative path.
     // ex) d:/root/.env/lib/site-packages, d:/root/.env
-    roots = roots
-        .map((r) => ensureTrailingDirectorySeparator(r))
-        .sort((a, b) => a.localeCompare(b))
-        .reverse();
+    roots = roots.sort((a, b) => a.key.localeCompare(b.key)).reverse();
 
     const separatorRegExp = /[\\/]/g;
-    function printFileOrModuleName(filePathOrModule: string | AbsoluteModuleDescriptor | undefined) {
-        if (filePathOrModule) {
-            if (typeof filePathOrModule === 'string') {
+    function printFileOrModuleName(fileUriOrModule: Uri | AbsoluteModuleDescriptor | undefined) {
+        if (fileUriOrModule) {
+            if (Uri.isUri(fileUriOrModule)) {
                 for (const root of roots) {
-                    if (filePathOrModule.startsWith(root)) {
-                        const subFile = filePathOrModule.substring(root.length);
-                        return stripFileExtension(subFile).replace(separatorRegExp, '.');
+                    if (fileUriOrModule.isChild(root)) {
+                        const subFile = root.getRelativePath(fileUriOrModule);
+                        return subFile!.split('.')[0].replace(separatorRegExp, '.');

```

Comment:
Should we continue to use the `stripFileExtension` from `pathUtils` for this?

Reply:
Yes another good catch.